InvenTree/InvenTree/ready.py

```diff
@@ -60,3 +73,21 @@ def canAppAccessDatabase(allow_test: bool = False, allow_plugins: bool = False,
             return False
 
     return True
+
+
+def isPluginRegistryLoaded():
+    """The plugin registry reloads all apps onetime after starting so that the discovered AppConfigs are added to Django.
+
+    This triggeres the ready function of AppConfig to execute twice. Add this check to prevent from running two times.
+
+    Returns: 'False' if the apps have not been reloaded already to prevent running the ready function twice
+    """
+    from django.conf import settings
+
+    # If plugins are not enabled, there wont be a second load
+    if not settings.PLUGINS_ENABLED:
+        return True
+
+    from plugin import registry
+
+    return not registry.is_loading
```

Comment:
Do you think that has some unwanted behavior if we include that everywhere where we have that canAppAccessDatabase also the isInMainThread Check from above in this file?

Reply:
The only area that could be problematic would be tasks (as they always run in the background). Would probably be an improvement overall

Reply:
Should I add those two functions to all other app configs or should I move these two checks to the canAppAccessDatabase function?

(Sure I'll do this in a separate pr)

Reply:
I would keep it as a separate function but add it to all configs.

Reply:
But should I combine the isInMainThread and isPluginRegistryLoaded function?

Reply:
I would keep them separate, they address different questions. Is `isMainThread` working now?

Reply:
Oh right, have to check if isMainThread works first?